zerver/middleware.py

```diff
@@ -737,3 +737,26 @@ def process_request(self, request: HttpRequest) -> HttpResponse | None:
             return response
 
         return None
+
+
+class InvalidRequestError(JsonableError):
+    http_status_code: int = 403
+    code: ErrorCode = ErrorCode.INVALID_REQUEST
```

Comment:
I think we should scope this error code more narrowly, and have a more specific name.  "invalid request" is _very_ broad, and we're not using it in many other cases where the request is invalid.

Reply:
Updated to use BadAPIRequest and BadAPIRequestError.